### PR TITLE
HTML: test coverage 7.1.5: `_top`, `_self`, existing contexts

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR is the final set of related test-coverage work on 7.1.5 Browsing Context Names.

This set of tests pertains to:

* `_top` named context
* `_self` named context
* The selection of existing contexts

It also takes care of some needed housekeeping and organization.

Unsurprisingly, the case-sensitivity issue for Webkit, Chrome is manifest again in `_top` and `_self`; I'll open bugs accordingly.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/5233)

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5233/)

**Comment by lyzadanger** [comment]
`_top`, `_self` case-sensitivity:

* Webkit bug: https://bugs.webkit.org/show_bug.cgi?id=170128
* Chromium bug: https://bugs.chromium.org/p/chromium/issues/detail?id=705646

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/215656435)

# Firefox (nightly channel) #
Testing web-platform-tests at revision abe54d7bf90939ace4fc5378131f764ede2ba884
Using browser at version BuildID 20170326110231; SourceStamp f5e214144799889e2408c4841351f4053f00544e
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>15 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_blank-001.html">/html/browsers/windows/browsing-context-names/choose-_blank-001.html</a></summary>


|                                 Subtest                                  | Results | Messages |
|--------------------------------------------------------------------------|---------|----------|
|                                                                          | OK      |          |
| `window.open into _blank should create a new browsing context each time` | PASS    |          |
| `_blank should be ASCII case-insensitive`                                | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_blank-002.html">/html/browsers/windows/browsing-context-names/choose-_blank-002.html</a></summary>


|                                          Subtest                                           | Results | Messages |
|--------------------------------------------------------------------------------------------|---------|----------|
|                                                                                            | OK      |          |
| `Context for opened noreferrer link targeted to "_blank" should not have opener reference` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_blank-003.html">/html/browsers/windows/browsing-context-names/choose-_blank-003.html</a></summary>


|                                   Subtest                                   | Results | Messages |
|-----------------------------------------------------------------------------|---------|----------|
|                                                                             | OK      |          |
| `Context created by link targeting "_blank" should retain opener reference` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_parent-001.html">/html/browsers/windows/browsing-context-names/choose-_parent-001.html</a></summary>


|                                  Subtest                                  | Results | Messages |
|---------------------------------------------------------------------------|---------|----------|
|                                                                           | OK      |          |
| `The parent browsing context must be chosen if the given name is _parent` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_parent-002.html">/html/browsers/windows/browsing-context-names/choose-_parent-002.html</a></summary>


|                       Subtest                        | Results | Messages |
|------------------------------------------------------|---------|----------|
|                                                      | OK      |          |
| `choosing _parent context: multiple nested contexts` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_parent-003.html">/html/browsers/windows/browsing-context-names/choose-_parent-003.html</a></summary>


|                   Subtest                    | Results |     Messages     |
|----------------------------------------------|---------|------------------|
|                                              | TIMEOUT |                  |
| `_parent should reuse window.parent context` | TIMEOUT | `Test timed out` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_parent-004.html">/html/browsers/windows/browsing-context-names/choose-_parent-004.html</a></summary>


|                        Subtest                        | Results | Messages |
|-------------------------------------------------------|---------|----------|
|                                                       | OK      |          |
| `choosing _parent context should be case-insensitive` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_self-001.html">/html/browsers/windows/browsing-context-names/choose-_self-001.html</a></summary>


|                                  Subtest                                   | Results | Messages |
|----------------------------------------------------------------------------|---------|----------|
|                                                                            | OK      |          |
| `The current browsing context must be chosen if the given name is "_self"` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_self-002.html">/html/browsers/windows/browsing-context-names/choose-_self-002.html</a></summary>


|                       Subtest                       | Results | Messages |
|-----------------------------------------------------|---------|----------|
|                                                     | OK      |          |
| `choosing _self context should be case-insensitive` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_top-001.html">/html/browsers/windows/browsing-context-names/choose-_top-001.html</a></summary>


|                                Subtest                                | Results | Messages |
|-----------------------------------------------------------------------|---------|----------|
|                                                                       | OK      |          |
| `Should choose current browsing context for "_top" if current is top` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_top-002.html">/html/browsers/windows/browsing-context-names/choose-_top-002.html</a></summary>


|                                Subtest                                | Results | Messages |
|-----------------------------------------------------------------------|---------|----------|
|                                                                       | OK      |          |
| `Should choose top browsing context for "_top" if current is not top` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_top-003.html">/html/browsers/windows/browsing-context-names/choose-_top-003.html</a></summary>


|                      Subtest                       | Results | Messages |
|----------------------------------------------------|---------|----------|
|                                                    | OK      |          |
| `choosing _top context should be case-insensitive` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-default-001.html">/html/browsers/windows/browsing-context-names/choose-default-001.html</a></summary>


|                                                      Subtest                                                       | Results |                                                                                                                                                                                                                                                                                                                                                                                Messages                                                                                                                                                                                                                                                                                                                                                                               |
|--------------------------------------------------------------------------------------------------------------------|---------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                                                    | OK      |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| `A embedded browsing context has empty-string default name`                                                        | PASS    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| `A browsing context which is opened by window.open() method with '_blank' parameter has empty-string default name` | PASS    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| `A browsing context has an empty-string default name`                                                              | FAIL    | `assert_equals: expected "" but got "d208530f-3697-46ed-9292-a1d280783d81";assert_equals: expected "" but got "adb66ab9-9c3c-4052-9eb2-ddb43461fab0";assert_equals: expected "" but got "fc8f00bb-a2b2-420f-9d93-638d8f7e403e";assert_equals: expected "" but got "3a9cc2a9-56b0-4f8b-b475-911f187b1179";assert_equals: expected "" but got "70ea815d-039b-40bc-b797-f5cdd7fd7885";assert_equals: expected "" but got "9077b74a-103b-44e3-82f6-96bb56d4b710";assert_equals: expected "" but got "9fa3fccf-4c43-4f2b-804a-930f61019a96";assert_equals: expected "" but got "1bdc8530-1b67-453f-b6e5-e938f740ba17";assert_equals: expected "" but got "c3e531d0-48e5-4b08-a175-003ebb3b5730";assert_equals: expected "" but got "9e3cf826-fa39-47aa-b290-7c2845fdc138"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-default-002.html">/html/browsers/windows/browsing-context-names/choose-default-002.html</a></summary>


|                                     Subtest                                     | Results | Messages |
|---------------------------------------------------------------------------------|---------|----------|
|                                                                                 | OK      |          |
| `The current browsing context must be chosen if the given name is empty string` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-existing-001.html">/html/browsers/windows/browsing-context-names/choose-existing-001.html</a></summary>


|                                         Subtest                                         | Results | Messages |
|-----------------------------------------------------------------------------------------|---------|----------|
|                                                                                         | OK      |          |
| `An existing browsing context must be chosen if the given name is the same as its name` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/215656436)

# Chrome (unstable channel) #
Testing web-platform-tests at revision abe54d7bf90939ace4fc5378131f764ede2ba884
Using browser at version 59.0.3047.0 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>15 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_blank-001.html">/html/browsers/windows/browsing-context-names/choose-_blank-001.html</a></summary>


|                                 Subtest                                  | Results |                              Messages                              |
|--------------------------------------------------------------------------|---------|--------------------------------------------------------------------|
|                                                                          | OK      |                                                                    |
| `window.open into _blank should create a new browsing context each time` | PASS    |                                                                    |
| `_blank should be ASCII case-insensitive`                                | FAIL    | `assert_not_equals: got disallowed value object "[object Window]"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_blank-002.html">/html/browsers/windows/browsing-context-names/choose-_blank-002.html</a></summary>


|                                          Subtest                                           | Results | Messages |
|--------------------------------------------------------------------------------------------|---------|----------|
|                                                                                            | OK      |          |
| `Context for opened noreferrer link targeted to "_blank" should not have opener reference` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_blank-003.html">/html/browsers/windows/browsing-context-names/choose-_blank-003.html</a></summary>


|                                   Subtest                                   | Results | Messages |
|-----------------------------------------------------------------------------|---------|----------|
|                                                                             | OK      |          |
| `Context created by link targeting "_blank" should retain opener reference` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_parent-001.html">/html/browsers/windows/browsing-context-names/choose-_parent-001.html</a></summary>


|                                  Subtest                                  | Results | Messages |
|---------------------------------------------------------------------------|---------|----------|
|                                                                           | OK      |          |
| `The parent browsing context must be chosen if the given name is _parent` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_parent-002.html">/html/browsers/windows/browsing-context-names/choose-_parent-002.html</a></summary>


|                       Subtest                        | Results | Messages |
|------------------------------------------------------|---------|----------|
|                                                      | OK      |          |
| `choosing _parent context: multiple nested contexts` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_parent-003.html">/html/browsers/windows/browsing-context-names/choose-_parent-003.html</a></summary>


|                   Subtest                    | Results | Messages |
|----------------------------------------------|---------|----------|
|                                              | OK      |          |
| `_parent should reuse window.parent context` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_parent-004.html">/html/browsers/windows/browsing-context-names/choose-_parent-004.html</a></summary>


|                        Subtest                        | Results |                     Messages                     |
|-------------------------------------------------------|---------|--------------------------------------------------|
|                                                       | OK      |                                                  |
| `choosing _parent context should be case-insensitive` | FAIL    | `assert_equals: expected "false" but got "true"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_self-001.html">/html/browsers/windows/browsing-context-names/choose-_self-001.html</a></summary>


|                                  Subtest                                   | Results | Messages |
|----------------------------------------------------------------------------|---------|----------|
|                                                                            | OK      |          |
| `The current browsing context must be chosen if the given name is "_self"` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_self-002.html">/html/browsers/windows/browsing-context-names/choose-_self-002.html</a></summary>


|                       Subtest                       | Results |                     Messages                     |
|-----------------------------------------------------|---------|--------------------------------------------------|
|                                                     | OK      |                                                  |
| `choosing _self context should be case-insensitive` | FAIL    | `assert_equals: expected "false" but got "true"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_top-001.html">/html/browsers/windows/browsing-context-names/choose-_top-001.html</a></summary>


|                                Subtest                                | Results | Messages |
|-----------------------------------------------------------------------|---------|----------|
|                                                                       | OK      |          |
| `Should choose current browsing context for "_top" if current is top` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_top-002.html">/html/browsers/windows/browsing-context-names/choose-_top-002.html</a></summary>


|                                Subtest                                | Results | Messages |
|-----------------------------------------------------------------------|---------|----------|
|                                                                       | OK      |          |
| `Should choose top browsing context for "_top" if current is not top` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-_top-003.html">/html/browsers/windows/browsing-context-names/choose-_top-003.html</a></summary>


|                      Subtest                       | Results |                      Messages                     |
|----------------------------------------------------|---------|---------------------------------------------------|
|                                                    | OK      |                                                   |
| `choosing _top context should be case-insensitive` | FAIL    | `assert_equals: expected "topWin" but got "_ToP"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-default-001.html">/html/browsers/windows/browsing-context-names/choose-default-001.html</a></summary>


|                                                      Subtest                                                       | Results |                                                                                                                                                                                                                                                                                                                                                                                Messages                                                                                                                                                                                                                                                                                                                                                                               |
|--------------------------------------------------------------------------------------------------------------------|---------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                                                    | OK      |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| `A embedded browsing context has empty-string default name`                                                        | PASS    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| `A browsing context which is opened by window.open() method with '_blank' parameter has empty-string default name` | PASS    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| `A browsing context has an empty-string default name`                                                              | FAIL    | `assert_equals: expected "" but got "4ef943a5-9046-4cdf-b227-0cdc01c1892e";assert_equals: expected "" but got "36e1d778-0bae-445b-b8ec-6e0dd7132594";assert_equals: expected "" but got "2ac51f7c-f4f9-4f17-9020-0755bba27cb7";assert_equals: expected "" but got "16d27f85-9944-4dfe-b7a0-e41b45c4e9ae";assert_equals: expected "" but got "222a1444-81d1-4720-b955-f1fe95584fb9";assert_equals: expected "" but got "d1fa49bf-8648-490a-9592-0d9d96e6d4d0";assert_equals: expected "" but got "0d26308e-dc27-4b1e-9e81-0a99cb28f47d";assert_equals: expected "" but got "4ba85899-9e25-42d1-b8d1-22bc93a1b676";assert_equals: expected "" but got "30102798-b886-45d5-92a1-8f5ad5e2f5bd";assert_equals: expected "" but got "e4aee36c-f457-46a7-a943-8f3d5e207366"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-default-002.html">/html/browsers/windows/browsing-context-names/choose-default-002.html</a></summary>


|                                     Subtest                                     | Results | Messages |
|---------------------------------------------------------------------------------|---------|----------|
|                                                                                 | OK      |          |
| `The current browsing context must be chosen if the given name is empty string` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5233/html/browsers/windows/browsing-context-names/choose-existing-001.html">/html/browsers/windows/browsing-context-names/choose-existing-001.html</a></summary>


|                                         Subtest                                         | Results | Messages |
|-----------------------------------------------------------------------------------------|---------|----------|
|                                                                                         | OK      |          |
| `An existing browsing context must be chosen if the given name is the same as its name` | PASS    |          |

</details>
</details>

File: html/browsers/windows/browsing-context-names/choose-_parent-004.html [hunk 1]
```diff
@@ -3,14 +3,38 @@
 <title>HTML Test: Choose browsing context - '_parent' (case-sensitivity)</title>
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
+<script src="/common/PrefixedLocalStorage.js"></script>
+<body>
 <div id="log"></div>
-<iframe src="resources/choose-_parent-004-iframe-1.html" name="parentWin" style="display:none"></iframe>
+</body>
```
Comment (zcorpan) [inline html/browsers/windows/browsing-context-names/choose-_parent-004.html]: Remove this line (or move down)

File: html/browsers/windows/browsing-context-names/resources/choose-_parent-004-iframe-1.html [hunk 2]
```diff
@@ -1,4 +1,14 @@
 <!DOCTYPE html>
 <meta charset="utf-8">
 <title>HTML Test: browsing context name - parent</title>
-<iframe src="choose-_parent-004-iframe-2.html"></iframe>
+<script src="/common/PrefixedLocalStorage.js"></script>
+<body>
+</body>
```
Comment (zcorpan) [inline html/browsers/windows/browsing-context-names/resources/choose-_parent-004-iframe-1.html]: Remove (or move down)

**Comment by zcorpan** [comment]
I can't reproduce this failure in Firefox (for me it passes):

/html/browsers/windows/browsing-context-names/choose-default-001.html

> A browsing context has an empty-string default name	FAIL	assert_equals: expected "" but got "d208530f-3697-46ed-9292-a1d280783d81";assert_equals: expected "" but got "adb66ab9-9c3c-4052-9eb2-ddb43461fab0";assert_equals: expected "" but got "fc8f00bb-a2b2-420f-9d93-638d8f7e403e";assert_equals: expected "" but got "3a9cc2a9-56b0-4f8b-b475-911f187b1179";assert_equals: expected "" but got "70ea815d-039b-40bc-b797-f5cdd7fd7885";assert_equals: expected "" but got "9077b74a-103b-44e3-82f6-96bb56d4b710";assert_equals: expected "" but got "9fa3fccf-4c43-4f2b-804a-930f61019a96";assert_equals: expected "" but got "1bdc8530-1b67-453f-b6e5-e938f740ba17";assert_equals: expected "" but got "c3e531d0-48e5-4b08-a175-003ebb3b5730";assert_equals: expected "" but got "9e3cf826-fa39-47aa-b290-7c2845fdc138"

Do you have a clue?

**Review by zcorpan — CHANGES_REQUESTED** [review]
LGTM % nits and uncertainty with a result in Firefox

**Comment by lyzadanger** [comment]
@zcorpan 

> I can't reproduce this failure in Firefox (for me it passes):

> /html/browsers/windows/browsing-context-names/choose-default-001.html

> Do you have a clue?

I have a clue...ish. This is one of the tests that I inherited, and I had observed the behavior you're seeing, too. For posterity, here's the test, which is _super_ basic:

```
//This test must be run when the current browsing context's name is not set
test(t => {
  assert_equals(window.name, "");
}, "A browsing context has an empty-string default name");
```

(Comments pre-exist my changes, also).

This test will pass when run directly and individually in a browser, but will fail when the test is run in a window opened (and ostensibly named) by a test runner. e.g. if I run the whole suite of tests in `browsing-context-names` in a browser (any browser, in my experience, not just FF):

image

it will fail.

I'm not sure of the best course of action here. I don't think the test adds a whole lot of value, and the default-window-name-is-empty-string notion is tested elsewhere (e.g. line 13 of the same test file). So part of me wants to yank it altogether.

Another part of me thinks that removing it is just patching over the fact that I'm not sure how to get it to work correctly in the test harnesses we've got, which is not ideal :).

So I'm not sure! Thoughts?

**Comment by jgraham** [comment]
Does the test work if you first `window.open` a new window and run the test in that?

**Comment by zcorpan** [comment]
Yeah that's already tested as well in the same file:
https://github.com/lyzadanger/web-platform-tests/blob/ece4f79e73629689bf99096fddef84160f3bcda9/html/browsers/windows/browsing-context-names/choose-default-001.html#L20

I suppose we should drop that part of the test. But it seems a bit limiting to not be able to assert `window.name` without creating a separate browsing context.

**Comment by lyzadanger** [comment]
Welp, for now, I've removed that test!

**Comment by cdumez** [comment]
Is this in landing shape now? Looking forward to getting more coverage in this area 👍

**Review by zcorpan — APPROVED** [review]

**Comment by zcorpan** [comment]
Yep! But should try to get CI checks green before landing. @lyzadanger can you squash and rebase on latest master and see if the bots are happy then?

**Pull request merged**